Product/Production/Gateway/DocumentSubmission_20/src/main/java/gov/hhs/fha/nhinc/docsubmission/_20/entity/deferred/response/EntityDocSubmissionDeferredResponseImpl_g1.java

```diff
@@ -59,10 +65,26 @@ public XDRAcknowledgementType provideAndRegisterDocumentSetBResponse(
             WebServiceContext context) {
         AssertionType assertion = getAssertion(context, provideAndRegisterDocumentSetAsyncRespRequest.getAssertion());
 
-        XDRAcknowledgementType response = outboundDocSubmissionResponse.provideAndRegisterDocumentSetBAsyncResponse(
+        return provideAndRegisterDocumentSetResponse(
                 provideAndRegisterDocumentSetAsyncRespRequest.getRegistryResponse(), assertion,
                 provideAndRegisterDocumentSetAsyncRespRequest.getNhinTargetCommunities());
-
+    }
+    /**
+     *
+     * @param request
+     * @param assertion
+     * @param targets
+     * @return
+     */
+    private XDRAcknowledgementType provideAndRegisterDocumentSetResponse(RegistryResponseType request,
+            AssertionType assertion, NhinTargetCommunitiesType targets){
+        XDRAcknowledgementType response = null;
+        try{
+            DocSubmissionUtils.getInstance().setTargetCommunitiesVersion(targets, UDDI_SPEC_VERSION.SPEC_2_0);
+            response = outboundDocSubmissionResponse.provideAndRegisterDocumentSetBAsyncResponse(request,assertion,targets);
+        }catch(Exception e){
+            LOG.error("Failed to send request to Nwhin ",e);
```

Comment:
Can you log exceptions using format ("message: {}", e.getLocalizedMessage(), e)?